### PR TITLE
Send a strong signal to delete feed item memberships when deleting a Webapp instance to prevent them from getting garbage collected (bug 1124319).

### PR DESCRIPTION
**Pull request by chuckharmston**
This was a fun one.

### CODE REVIEW

File: mkt/feed/tests/test_serializers.py [hunk 1]
```diff
@@ -284,6 +284,19 @@ def test_is_published(self):
...
         assert data['is_published']
 
+    def test_shelf_get_apps_deleted_app(self):
+        """
+        Regression test for bug 1124319; ensures that deleted apps also delete
+        any respective FeedShelfMembership objects.
+        """
+        app_id = self.app_ids.pop()
+        Webapp.objects.get(pk=app_id).delete()
+        eq_(FeedShelfMembership.objects.filter(app__id=app_id).count(), 0)
```
Comment (ngokevin) [inline mkt/feed/tests/test_serializers.py]: Can we assert this is 1 beforehand?

**Comment by ngokevin** [comment]
r+

File: mkt/feed/tests/test_serializers.py (hunk 1 above)
Comment (chuckharmston) [inline mkt/feed/tests/test_serializers.py]: Done.

**Comment by chuckharmston** [comment]
Thanks!

File: mkt/feed/models.py [hunk 2]
```diff
@@ -474,13 +474,11 @@ def delete_search_index(sender, instance, **kw):
...
+                FeedShelfMembership]:
         cls.objects.filter(app_id=instance.pk).delete()
-    return inner
 
-for cls in [FeedBrandMembership, FeedCollectionMembership,
-            FeedShelfMembership]:
-    post_delete.connect(remove_deleted_app_on(cls), sender=Webapp,
-                        dispatch_uid='apps_collections_cleanup')
+post_delete.connect(remove_memberships, sender=Webapp, weak=False,
+                    dispatch_uid='cleanup_feed_membership')
```
Comment (robhudson) [inline mkt/feed/models.py]: That makes more sense. This is one signal checking 3 model classes instead of the other way around.

**Pull request merged**